### PR TITLE
Support concurrent export of query results

### PR DESCRIPTION
**Pull request by EmmyMiao87**
## Proposed changes

This pr mainly supports
1. Export query result sets concurrently
2. Query result set export supports s3 protocol

Among them, there are several preconditions for concurrently exporting query result sets
1. Enable concurrent export variables
2. The query itself can be exported concurrently
    (some queries containing sort nodes at the top level cannot be exported concurrently)
3. Export the s3 protocol used instead of the broker

After exporting the result set concurrently,
the file prefix is changed to outfile_{query_instance_id}_filenumber.{file_format}

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: be/src/runtime/file_result_writer.h [hunk 1]
```diff
@@ -31,6 +32,7 @@ class RuntimeProfile;
 class TupleRow;
 
 struct ResultFileOptions {
+    // deprecated
     bool is_local_file;
```
Comment (morningman) [inline be/src/runtime/file_result_writer.h:36]: Why not delete it?

File: be/src/runtime/file_result_writer.cpp [hunk 2]
```diff
@@ -39,13 +42,36 @@ namespace doris {
...
           _output_expr_ctxs(output_expr_ctxs),
           _parent_profile(parent_profile),
-          _sinker(sinker) {}
+          _sinker(sinker) {
+        if (_file_opts->is_local_file) {
+            _storage_type = TStorageBackendType::LOCAL;
+        } else {
+            _storage_type = TStorageBackendType::BROKER;
+        }
+        _fragment_instance_id.hi = 12345678987654321;
```
[the remark this answers is not in the document]
Comment (EmmyMiao87) [reply be/src/runtime/file_result_writer.cpp:58]: In order to be compatible with the old version of Fe and the new version of be

File: be/src/runtime/file_result_writer.h (hunk 1 above)
Comment (EmmyMiao87) [reply be/src/runtime/file_result_writer.h:36]: For compatible

File: be/src/runtime/file_result_writer.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (morningman) [reply be/src/runtime/file_result_writer.cpp:58]: Add comment to explain.

BTW, I suggest to add `FileNamePrefix` column in the result of outfile operation. For easy to get full name of exported files. Such as:

```
+------------+-----------+-----------+-------------+------------------------------------------+
| FileNumber | TotalRows | FileSize  | URL         |FileNamePrefix                            |
+------------+-----------+-----------+-------------+------------------------------------------+
|          1 |    123605 | 361061014 | 127.0.0.1   |my_file2_46e9ef9b66924a79-92f887d43be10325|
|          1 |    128180 | 374334318 | 127.0.0.1   |my_file2_46e9ef9b66924a79-92f887d43be10326|
|          1 |    125156 | 365569023 | 127.0.0.1   |my_file2_46e9ef9b66924a79-92f887d43be10327|
|          1 |    124096 | 362395588 | 127.0.0.1   |my_file2_46e9ef9b66924a79-92f887d43be10328|
|          1 |    124862 | 364727515 | 127.0.0.1   |my_file2_46e9ef9b66924a79-92f887d43be10329|
|          1 |    124520 | 363649600 | 127.0.0.1   |my_file2_46e9ef9b66924a79-92f887d43be10330|
|          1 |    124447 | 363479285 | 127.0.0.1   |my_file2_46e9ef9b66924a79-92f887d43be10331|
|          1 |    125134 | 365490611 | 127.0.0.1   |my_file2_46e9ef9b66924a79-92f887d43be10332|
+------------+-----------+-----------+-------------+------------------------------------------+
```

[the remark this answers is not in the document]
Comment (EmmyMiao87) [reply be/src/runtime/file_result_writer.cpp:58]: It's not easy to add, there will be compatibility issues

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**